core/src/main/java/org/infinispan/configuration/cache/QueryConfigurationBuilder.java

```diff
@@ -35,6 +36,22 @@ public int defaultMaxResults() {
       return attributes.attribute(DEFAULT_MAX_RESULTS).get();
    }
 
+   /**
+    * Limit the required accuracy of the hit count for the indexed queries to an upper-bound.
+    * Setting the hit-count-accuracy could improve the performance of queries targeting large data sets.
```

Comment:
I think we should document what the expected possible values are for this. I have to admit I wasn't expecting 10,000 as a value...

Reply:
Good idea. I going to add the sentence: "For optimal performances set this value not much above the expected hit count. If you do not require accurate hit counts, set it to a low value."